src/System.Linq/tests/ConcatTests.cs

```diff
@@ -251,20 +251,34 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             var supposedlyLargeCollection = new DelegateBasedCollection<int> { CountWorker = () => int.MaxValue };
             var tinyCollection = new DelegateBasedCollection<int> { CountWorker = () => 1 };
 
+            Action<Action> assertThrows = (testCode) =>
+            {
+                // The full .NET framework uses unsigned arithmetic summing up collection counts.
+                // See https://github.com/dotnet/corefx/pull/6131.
+                if (PlatformDetection.IsFullFramework)
+                {
+                    testCode();
+                }
+                else
+                {
+                    Assert.Throws<OverflowException>(testCode);
```

Comment:
There is no point spending this much time validating that the behavior has not changed on the desktop.
I.E. It makes sense to have shimmed tests that expect that desktop behaves differently, but not when they take minutes to run.

If we are testing something that is an expected change or optimization and full FX takes a long time in that scenario, it feels fine to skip it.

---
In reply to: [109825089](https://github.com/dotnet/corefx/pull/17877#discussion_r109825089) [](ancestors = 109825089)